### PR TITLE
[#WIP] [Platform] [API] Instance Type controller error handling.

### PR DESCRIPTION
**Pull request by jitendra-12113**
**Description:**
Use the new global exception handling in the Instance Type controller.

Test Plan:

Its mostly refactor. Unit tests are updated.
Run server and test basic operations.

### CODE REVIEW

**Review by sb-yb — CHANGES_REQUESTED** [review]

**Comment by sb-yb** [comment]
Add [API] to subject

File: managed/src/main/java/com/yugabyte/yw/controllers/InstanceTypeController.java [hunk 1]
```diff
@@ -117,16 +115,12 @@ private Result maybeFilterByZoneOfferings(Set<String> filterByZoneCodes, Provide
...
-    Form<InstanceType> formData = formFactory.form(InstanceType.class).bindFromRequest();
+    Form<InstanceType> formData = formFactory.getFormDataOrBadRequest(InstanceType.class);
     if (formData.hasErrors()) {
-      return ApiResponse.error(BAD_REQUEST, formData.errorsAsJson());
-    }
-
-    Provider provider = Provider.get(customerUUID, providerUUID);
-    if (provider == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Provider UUID: " + providerUUID);
+      throw new YWServiceException(BAD_REQUEST, formData.errorsAsJson());
```
[the remark this answers is not in the document]
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/InstanceTypeController.java:120]: Yeah correct. Thanks for catching this.

File: managed/src/main/java/com/yugabyte/yw/controllers/InstanceTypeController.java [hunk 2]
```diff
@@ -170,7 +157,7 @@ public Result delete(UUID customerUUID, UUID providerUUID, String instanceTypeCo
       return ApiResponse.success(responseJson);
     } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/InstanceTypeController.java]: How about logging the exception that will also be gone post removal of try catch. 
 
 Though I removed them just thought to highlight. Thanks!

[the remark this answers is not in the document]
Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/controllers/InstanceTypeController.java]: It will get logged at top level handler.

**Review by sb-yb — APPROVED** [review]
Thanks for making the changes.
This looks good now!

**Pull request merged**